src/components/course/EnrollModal.jsx

```diff
@@ -81,22 +247,41 @@ const EnrollModal = ({
   }
 
   return (
-    <Modal
-      open={isModalOpen}
-      closeText="Cancel"
+    <AlertModal
+      isOpen={isModalOpen}
+      closeLabel={intl.formatMessage(messages.modalCancelCta)}
       title={titleText}
-      body={<div><p>{enrollText}</p></div>}
-      buttons={[
-        <Button
-          href={enrollmentUrl}
-          onClick={handleEnroll}
-        >
-          {isLoading && <Spinner animation="border" className="mr-2" variant="light" size="sm" screenReaderText="Loading" />}
-          {buttonText}
-        </Button>,
-      ]}
-      onClose={() => setIsModalOpen(false)}
-    />
+      footerNode={(
+        <ActionRow>
+          <Button
+            variant="tertiary"
+            onClick={dismissModal}
+          >
+            <FormattedMessage {...messages.modalCancelCta} />
+          </Button>
+          {/* FIXME: the following Button should be using StatefulButton from @openedx/paragon */}
+          <Button
+            // TODO: remove no-op behavior for learner credit
+            href={userSubsidyApplicableToCourse.subsidyType === LEARNER_CREDIT_SUBSIDY_TYPE ? undefined : enrollmentUrl}
```

Comment:
[inform] Gates the `EnrollModal`'s confirmation CTA to be a no-op (i.e., `undefined` vs. `enrollmentUrl` when the applicable subsidy type is `LEARNER_CREDIT_SUBSIDY_TYPE`. This `EnrollModal` component will not render/be opened if the API call to `can-redeem` (gated) is not made).

Part 2 of ENT-9066 will actually integrate this `Button` with the `StatefulEnroll` logic to trigger redemption.